inspirehep/search/factories/search.py

```diff
@@ -16,13 +16,17 @@
 
 def get_search_with_source(search):
     source = current_app.config.get("SEARCH_SOURCE_INCLUDES")
-    if not source:
+    has_accept_mimetypes = len(request.accept_mimetypes) > 0
+
+    if not source or not has_accept_mimetypes:
         return search
 
     if isinstance(search, LiteratureSearch):
-        source_literature = source.get("literature")
-        if source_literature:
-            search = search.source(includes=source_literature)
+        source_literature = source.get("literature", {})
+        accept_type = next(request.accept_mimetypes.values())
+        source_literature_mimetype = source_literature.get(accept_type)
+        if source_literature_mimetype:
+            search = search.source(includes=source_literature_mimetype)
```

Comment:
it's everything if `includes` is `None` right?

Reply:
yes, it will never get there with ``None`` though. It'll `` return search``